express-joi-validation.js

```diff
@@ -59,6 +59,19 @@ function buildErrorString(err, container) {
   return ret
 }
 
+function patchResponseForResponseValidation(res, validationFn) {
+  res._origJson = res.json.bind(res)
+  res.json = validationFn
+}
+
+function restoreResponseAfterResponseValidation(res) {
+  if (res._origJson) {
+    res.json = res._origJson
+  }
+}
+
+module.exports.fixupResponse = restoreResponseAfterResponseValidation
```

Comment:
I think it'd be best not to expose this. Make the fix-up automatic in all scenarios, including `response`, if `passError` is enabled. WDYT?

Reply:
Ah, I see your last comment. Right, this is a tricky...

Reply:
This is indeed Hacky. Do you have any other suggestion?

Reply:
I had this idea stewing in my head around providing schemas for certain status code ranges. It's a bit more complex though. Something like:

```js
validation.response({
  // Schemas to run for given status codes
  '2xx': schemaSuccess,
  '4xx': schemaFail
}, {
  // These could be defined globally instead of as part of 
  // the call to response()
  passError: true
  passErrorSchemas: {
    '5xx': errrorHandlerSchema
  }
})
```

If no schema is provided for 5xx, then perhaps validation is bypassed?

Reply:
This seems like a very good interface.
It seems to me that it will still need the mechanism of overriding the `json` function on the `Response` object and require restoring the original `json` function